hack/manifest-gen/internal/manifest.go

```diff
@@ -0,0 +1,117 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package internal
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+)
+
+// GenerateFlags holds flag values for the generate operation.
+type GenerateFlags struct {
+	Source      string
+	Profile     string
+	Values      []string
+	ValueFiles  []string
+	ExcludeCRDs bool
+}
+
+// OptionsFlags holds flag values for the options operation.
+type OptionsFlags struct {
+	Source string
+}
+
+// Generate produces a manifest for installing ECK using an embedded Helm chart.
+func Generate(opts *GenerateFlags) error {
```

Comment:
It'd be nice to have a few comments explaining how this relies on helm internally and differs (or not?) from what helm would normally do if you invoke it directly.